data-serving/data-service/src/model/location.ts

```diff
@@ -1,21 +1,29 @@
 import mongoose from 'mongoose';
 
+export enum GeoResolution {
+    Point = 'Point', // Lat/long coordinates
+    Admin3 = 'Admin3', // ~City
+    Admin2 = 'Admin2', // ~County
+    Admin1 = 'Admin1', // ~Province
+    Admin0 = 'Admin0', // Country
+}
+
 const fieldRequiredValidator = [
     function (this: LocationDocument): boolean {
         return (
             this != null &&
             this.country == null &&
             this.administrativeAreaLevel1 == null &&
             this.administrativeAreaLevel2 == null &&
-            this.locality == null
+            this.administrativeAreaLevel3 == null &&
+            this.place == null
         );
     },
     'One of country, administrativeAreaLevel1, administrativeAreaLevel2, ' +
-        'or locality is required',
+        'administrativeAreaLevel3, or place is required',
```

Comment:
Related to my prior question on the content of `place` -- is this "safe"? If `place` is a string, is it guaranteed to stand in for an administrative area or country?

Reply:
Maybe we should rename this to "name" and have it stand for a "human readable name of the finest grain location that this represent"?
(that would be the "text" in the mapbox api responses for example)